executor/aggfuncs/func_bit_or.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggfuncs
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+type baseBitAggFunc struct {
+	baseAggFunc
+}
+
+type bitOrUint64 struct {
+	baseBitAggFunc
+}
+
+type partialResult4BitFunc = uint64
+
+func (e *bitOrUint64) AllocPartialResult() PartialResult {
```

Comment:
This can be a member function of `*baseBitAggFunc`.

Reply:
great suggestion.

Reply:
...I take back the last sentence. 😂 
We should not make the result value to be a member of `baseBitAggFunc `,  Because this will make `The baseBitAggFunc` to be Stateful. 
Consider another scenario, If we have many groups to be aggregated, if the `AggFunc` is statefull, we have to create many `aggFunc` to handle this. 
But if `AggFunc` is not statefull, we can only create one `AggFunc` and many `partialResult4BitFunc`, this will reduce go GC pressure. 
( This is @zz-jason told me. Thanks very much~ )